service/src/main/java/uk/gov/hmcts/reform/fpl/service/ChildrenMapperService.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.model.Children;
+
+import java.util.Map;
+import java.util.Optional;
+
+@Service
+public class ChildrenMapperService {
+
+    private final ObjectMapper mapper;
+    private final Logger logger = LoggerFactory.getLogger(ChildrenMapperService.class);
+
+    @Autowired
+    public ChildrenMapperService(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    public Optional<Children> mapChildren(final Map<String, Object> childrenMap) {
+        try {
+            String json = mapper.writeValueAsString(childrenMap);
+            return Optional.of(mapper.readValue(json, Children.class));
+        } catch (Exception e) {
+            logger.error("Exception parsing children data " + e.toString());
+            return Optional.empty();
```

Comment:
I am agains returning empty objects / null in case there was exception. It might hide real issue for long time. IMO empty optional / null should have been returned only when input map (source) is empty. In case there is exception thrown it should be logged, re-thrown to break application if not handled outside.